nmtwizard/preprocess/tu.py

```diff
@@ -210,43 +210,71 @@ def __init__(self,
                  alignment=None,
                  source_tokenizer=None,
                  target_tokenizer=None):
-        self.__source = TranslationSide(source, tokenizer=source_tokenizer)
+        self.__source = [TranslationSide(source, tokenizer=source_tokenizer)]
         self.__target = None
         self.__metadata = metadata if metadata is not None else [None] #TODO: proper metadata
         self.__annotations = annotations
         self.__alignment = None
 
         if target is not None:
-            self.__target = TranslationSide(target, tokenizer=target_tokenizer)
+            self.__target = [TranslationSide(target, tokenizer=target_tokenizer)]
             if alignment is not None:
                 self.__alignment = Alignment(alignments=alignment)
 
+    def add_source(self, source, source_tokenizer=None):
+        self.__source.append(TranslationSide(source, tokenizer=source_tokenizer))
+
+    def add_target(self, target, target_tokenizer=None):
+        self.__target.append(TranslationSide(target, tokenizer=target_tokenizer))
+
+    @property
+    def source_num(self):
+        return len(self.__source)
+
+    @property
+    def target_num(self):
+        return len(self.__target) if self.__target is not None else 0
```

Comment:
These names would sound more natural to me in this context ("num" first and plural):

```suggestion
    @property
    def num_sources(self):
        return len(self.__source)

    @property
    def num_targets(self):
        return len(self.__target) if self.__target is not None else 0
```